tests/openssh.rs

```diff
@@ -1057,3 +1057,73 @@ async fn test_read_large_file_bug() {
         assert_eq!(stdout.len(), bs * count);
     }
 }
+
+#[tokio::test]
+#[cfg_attr(not(ci), ignore)]
+async fn arc_client() {
+    for session in connects().await {
+        let session = std::sync::Arc::new(session);
+        let mut child = session
+            .clone()
+            .arc_command("cat")
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .spawn()
+            .await
+            .unwrap();
+
+        drop(session);
+
+        // write something to standard in and send EOF
+        let mut stdin = child.stdin().take().unwrap();
+        stdin.write_all(b"hello world").await.unwrap();
+        drop(stdin);
+
+        // cat should print it back on stdout
+        let mut stdout = child.stdout().take().unwrap();
+        let mut out = String::new();
+        stdout.read_to_string(&mut out).await.unwrap();
+        assert_eq!(out, "hello world");
+        drop(stdout);
+
+        // cat should now have terminated
+        let status = child.wait().await.unwrap();
+
+        // ... successfully
+        assert!(status.success());
+    }
+}
+
+#[tokio::test]
+#[cfg_attr(not(ci), ignore)]
+async fn generic_client() {
```

Comment:
```suggestion
async fn rc_client() {
```

I think `rc_client` is a better name here.

Reply:
Dunno, the intent of the name was to indicate that we're testing the generic `Session::to_command` rather than the ref-specific Session::command` or the Arc-specific `Session::arc_command`, the Rc is an implementation detail of the test but I figured I might as well use that since references and arc are already exercised. Would it help if I made most of the body generic and then ran it with both Rc and Arc?

Reply:
It makes sense then, though I would like to rename this and the other test fn  to test_session_to_command and test_session_arc_command

Reply:
Works for me.